src/pages/RFOX/hooks/useGetUnstakingRequestQuery.ts

```diff
@@ -0,0 +1,100 @@
+import { skipToken } from '@tanstack/react-query'
+import { foxStakingV1Abi } from 'contracts/abis/FoxStakingV1'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
+import { useMemo } from 'react'
+import { type Address, getAddress, type MulticallReturnType } from 'viem'
+import { multicall } from 'viem/actions'
+import { arbitrum } from 'viem/chains'
+import type { Config } from 'wagmi'
+import { type ReadContractsQueryKey, useQuery } from 'wagmi/query'
+import { isSome } from 'lib/utils'
+import { viemClientByNetworkId } from 'lib/viem-client'
+
+import { useGetUnstakingRequestCountQuery } from './useGetUnstakingRequestCountQuery'
+
+type AllowFailure = false
+
+const getContracts = (stakingAssetAccountAddress: string | undefined, count: bigint) =>
+  stakingAssetAccountAddress
+    ? Array.from(
+        { length: Number(count) },
+        (_, index) =>
+          ({
+            abi: foxStakingV1Abi,
+            address: RFOX_PROXY_CONTRACT_ADDRESS,
+            functionName: 'getUnstakingRequest',
+            args: [getAddress(stakingAssetAccountAddress), BigInt(index)],
+            chainId: arbitrum.id,
+          }) as const,
+      )
+    : []
+
+type GetContractsReturnType = ReturnType<typeof getContracts>
+
+type GetUnstakingRequestQueryKey = ReadContractsQueryKey<[Address, bigint], AllowFailure, Config>
+
+type UnstakingRequest = MulticallReturnType<GetContractsReturnType, AllowFailure>
```

Comment:
These multicall types took some fandangling, but we got there!

Reply:
I can feel the pain, absolute 🧠